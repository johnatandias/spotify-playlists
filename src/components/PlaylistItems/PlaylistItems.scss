@import '../../variables';

.playlistItems {
  display: flex;
  flex-direction: column;
  margin-top: 30px;

  &__nameAndTime {
    display: flex;
    align-items: center;
    justify-content: space-between;

    &__name {
      color: $primary-text-color;
      font-weight: bold;

      &::before {
        display: inline-block;
        vertical-align: sub;
        width: 20px;
        content: "\2022";
        color: $highlight-color;
        font-size: 30px;
        font-weight: bold;
      }
    }

    &__time {
      color: $secondary-text-color;
      font-weight: 500;
    }
  }

  &__artist {
    color: $secondary-text-color;
    font-size: 10px;
    font-weight: 500;
    margin-left: 20px;
    margin-bottom: 15px;
  }
}
