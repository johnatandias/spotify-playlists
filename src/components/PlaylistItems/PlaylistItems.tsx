import React from 'react';
import './PlaylistItems.scss';

export interface PlaylistItem {
  name: string,
  artist: string,
  time: string
}

export const PlaylistItems = ({ items, className = '' }: { items: PlaylistItem[], className?: string }) => (
  <div className={`playlistItems ${className}`}>
    {items.map(({ name, artist, time }: PlaylistItem, index: number) => (
      <div key={index}>
        <div className="playlistItems__nameAndTime">
          <span className="playlistItems__nameAndTime__name">{name}</span>
          <span className="playlistItems__nameAndTime__time">{time}</span>
        </div>
        <span className="playlistItems__artist">{artist}</span>
      </div>
    ))}
  </div>
);
