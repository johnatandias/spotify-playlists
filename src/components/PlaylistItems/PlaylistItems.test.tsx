import React from "react";
import { shallow } from 'enzyme';
import { PlaylistItems, PlaylistItem } from './PlaylistItems';

describe('PlaylistItems component', () => {
  const items: PlaylistItem[] = [
    { name: 'name 1', artist: 'artist 1', time: '1:00' },
    { name: 'name 2', artist: 'artist 2', time: '2:00' },
    { name: 'name 3', artist: 'artist 3', time: '3:00' }
  ];

  it('Renders without crashing', () => {
    const wrapper = shallow(<PlaylistItems items={items} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Render all items', () => {
    const wrapper = shallow(<PlaylistItems items={items} />);
    expect(wrapper.children()).toHaveLength(3);
    expect(wrapper).toMatchSnapshot();
  });
});
