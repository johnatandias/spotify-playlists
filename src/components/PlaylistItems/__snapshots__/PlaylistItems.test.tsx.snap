// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistItems component Render all items 1`] = `ShallowWrapper {}`;

exports[`PlaylistItems component Renders without crashing 1`] = `ShallowWrapper {}`;
